lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java

```diff
@@ -0,0 +1,457 @@
+import pro.taskana.classification.api.models.Classification;
+import pro.taskana.common.api.TimeInterval;
+import pro.taskana.task.api.TaskCommentQuery;
+import pro.taskana.task.api.TaskCommentQueryColumnName;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.models.Task;
+import pro.taskana.task.api.models.TaskComment;
+import pro.taskana.testapi.DefaultTestEntities;
+import pro.taskana.testapi.TaskanaConfigurationModifier;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.builder.TaskBuilder;
+import pro.taskana.testapi.builder.TaskCommentBuilder;
+import pro.taskana.testapi.builder.WorkbasketAccessItemBuilder;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.user.api.UserService;
+import pro.taskana.user.api.models.User;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.exceptions.NotAuthorizedToQueryWorkbasketException;
+import pro.taskana.workbasket.api.models.Workbasket;
+
+@TaskanaIntegrationTest
+class QueryTaskCommentAccTest {
+
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject UserService userService;
+
+  Classification defaultClassification;
+  Workbasket defaultWorkbasket;
+  Task task1;
+  TaskComment comment1;
+  TaskComment comment2;
+  TaskComment comment3;
+
+  @WithAccessId(user = "user-1-1")
+  @BeforeEach
+  public void createTaskAndTaskComments() throws Exception {
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+    comment2 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text2")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+    comment3 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text3")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService, "admin");
+    List<TaskComment> taskComments = taskService.getTaskComments(task1.getId());
+    assertThat(taskComments).hasSize(3);
+  }
+
+  @WithAccessId(user = "admin")
+  @AfterEach
+  public void deleteTaskComments() throws Exception {
+    taskService.deleteTaskComment(comment1.getId());
+    taskService.deleteTaskComment(comment2.getId());
+    taskService.deleteTaskComment(comment3.getId());
+  }
+
+  @WithAccessId(user = "admin")
+  @BeforeAll
+  void setup() throws Exception {
+
+    User userWithName = userService.newUser();
+    userWithName.setId("user-1-1");
```

Comment:
Why do you add the user here instead of the nested classes that use that user?

Reply:
I have changed it so that the User will be initialized at the nested classes TaskCreatorFullNameWhenAdditionalUserInfoIsFalse, TaskCreatorFullNameWhenAdditionalUserInfoIsTrue and QueryColumnTaskComments

These are the 3 classes that query the long name. After all the tests in the nested class are executed, userservice will then delete the user "user-1-1"